pytorch_lightning/plugins/training_type/ipu.py

```diff
@@ -127,16 +126,24 @@ def pre_dispatch(self) -> None:
                 options=self.inference_opts,
             )
             self.poptorch_models[x] = model
+        self._handle_gradient_accumulation_steps()
 
     @property
-    def replication_factor(self):
+    def replication_factor(self) -> int:
+        # This should be set via the opts not the parallel devices
+        if self.lightning_module and self.lightning_module.trainer.training and self._training_opts is not None:
+            return self._training_opts.replication_factor
+        elif self._training_opts is not None:
```

Comment:
Not sure this logic is right here. Shouldn't this be: `self.lightning_module.trainer.training and self._training_opts is not None:`

Reply:
Yeah sounds right, I made it like this as the lightning module when this is called may not be a reference yet (we haven't called fit on the model). Thus we default to the training opts. Need to make this better though